.github/workflows/benchmarks.yml

```diff
@@ -143,6 +147,46 @@ jobs:
         with:
           path: asv-compare-changed-output.log
 
+      - name: Run benchmarks for base to head commits of PR
+        if: github.event_name == 'pull_request_target'
+        run: |
+          echo $(git rev-parse HEAD) > commit_hashes.txt
+          echo $(git merge-base HEAD upstream/master) >> commit_hashes.txt
+
+          asv run HASHFILE:commit_hashes.txt | tee asv-output-PR.log  
+            if grep -q failed asv-output-PR.log; then 
+              echo "Some benchmarks have failed!"
+              exit 1
+            fi
+
+      - name: Generate Graphs and HTML of PR
+        if: github.event_name == 'pull_request_target'
+        run: |
+          asv publish
+
+      - name: Delete env files of PR
+        if: github.event_name == 'pull_request_target'
+        run: rm -r .asv/env
+      
+      - name: Move .asv to PR folder
+        if: github.event_name == 'pull_request_target'
+        run: mv .asv PR-${{ github.event.number }}
+
+      - name: Push results of PR to results repository
+        if: github.event_name == 'pull_request_target'
+        continue-on-error: true
+        uses: cpina/github-action-push-to-another-repository@main 
+        env:
+          API_TOKEN_GITHUB: ${{ secrets.BOT_TOKEN }}
+        with:
+          source-directory: PR-${{ github.event.number }}
+          destination-github-username: tardis-sn
+          destination-repository-name: tardis-benchmarks
+          user-email: tardis.sn.bot@gmail.com
+          target-branch: main
+          target-directory: PR
+
```

Comment:
Have you tested this? This will rewrite a new folder called PR each time the workflow runs on any pull request. If I am not wrong, I doubt if this is the good way to publish a website. Suppose if there are two pull requests running the thing at the same time, then how would we know what pr the published website is referring to?

Reply:
I have added new changes to solve this.